src/main/java/io/prometheus/cloudwatch/CloudWatchCollector.java

```diff
@@ -304,10 +392,10 @@ public String getMonitoringEndpoint(Region region) {
     }
 
     /**
-     * Check if a metric should be used according to `aws_dimension_select` or `aws_dimension_select_regex`
+     * Check if a metric should be used according to `aws_dimension_select`, `aws_dimension_select_regex` or dynamic `aws_tag_select`
      */
-    private boolean useMetric(MetricRule rule, Metric metric) {
-      if (rule.awsDimensionSelect == null && rule.awsDimensionSelectRegex == null) {
+    private boolean useMetric(MetricRule rule, List<String> tagBasedResourceIds, Metric metric) {
```

Comment:
I'd expect this to be an and, not an or

Reply:
Currently, the logic is an `or` between aws_dimension_select and aws_dimension_select_regex.
Changing this to an `and` while adding aws_tag_select would break existing client behavior if they already use both aws_dimension_select and aws_dimension_select_regex in a same config bloc.
Do you think it worth changing to an `and` ?

Reply:
It doesn't make sense to use both aws_dimension_select and aws_dimension_select_regex, so I'd go for and.